adapters/coap/src/main/java/org/eclipse/hono/adapter/coap/option/TimeOption.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ * <p>
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ * <p>
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * <p>
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.adapter.coap.option;
+
+import org.eclipse.californium.core.coap.Option;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.coap.Response;
+
+/**
+ * CoAP custom time option.
+ * <p/>
+ * Used in {@link Request} to indicate that the client wants to get the servers system-time in milliseconds.
```

Comment:
It is great that you include extensive JavaDoc :+1: 
However, there is not much value in adding a link to the Cf `Request` class here because the JavaDoc tool will (usually) not be able to create a corresponding (HTML) link anyway since the Californium docs are not available (locally).

Instead, you might want to simply use
> Used in a CoAP request to indicate ...